File: app/src/main/java/com/Phase4/pathways/viewmodel/LoginRegisterViewModel.java
package com.Phase4.pathways.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.Phase4.pathways.model.AppRepository;
import com.google.firebase.auth.FirebaseUser;

public class LoginRegisterViewModel extends AndroidViewModel {
    private AppRepository appRepository;
    private MutableLiveData<FirebaseUser> userMutableLiveData;

    public LoginRegisterViewModel(@NonNull Application application) {
        super(application);

        appRepository = new AppRepository(application);
        userMutableLiveData = appRepository.getUserMutableLiveData();
    }

    // Register method
    public void register(String email, String password){
        appRepository.register(email, password);
    }

    // getter
    public MutableLiveData<FirebaseUser> getUserMutableLiveData() {
        return userMutableLiveData;
    }
}

File: app/src/main/java/com/Phase4/pathways/viewmodel/LoginRegisterViewModel.kt
package com.Phase4.pathways.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.Phase4.pathways.model.AppRepository
import androidx.lifecycle.MutableLiveData
import com.google.firebase.auth.FirebaseUser

class LoginRegisterViewModel(application: Application) : AndroidViewModel(application) {
    private val AppRepository: AppRepository
    val userLiveData: MutableLiveData<FirebaseUser>
    fun login(email: String?, password: String?) {
        AppRepository.login(email, password)
    }

    fun register(email: String?, password: String?) {
        AppRepository.register(email, password)
    }

    init {
        AppRepository = AppRepository(application)
        userLiveData = AppRepository.userLiveData
    }
}